plugins/pipelines/listBuilds.js

```diff
@@ -0,0 +1,77 @@
+'use strict';
+
+const boom = require('@hapi/boom');
+const joi = require('joi');
+const schema = require('screwdriver-data-schema');
+const buildListSchema = joi.array().items(schema.models.build.get).label('List of builds');
+const groupEventIdSchema = schema.models.event.base.extract('groupEventId');
+const pipelineIdSchema = schema.models.pipeline.base.extract('id');
+
+module.exports = () => ({
+    method: 'GET',
+    path: '/pipelines/{id}/builds',
+    options: {
+        description: 'Get builds for this pipeline',
+        notes: 'Returns builds for the given pipeline',
+        tags: ['api', 'pipelines', 'builds'],
+        auth: {
+            strategies: ['token'],
+            scope: ['user', 'build', 'pipeline']
+        },
+
+        handler: async (request, h) => {
+            const factory = request.server.app.pipelineFactory;
+
+            return factory
+                .get(request.params.id)
+                .then(pipeline => {
+                    if (!pipeline) {
+                        throw boom.notFound('Pipeline does not exist');
+                    }
+
+                    const config = {};
+
+                    if (request.query.page || request.query.count) {
+                        config.paginate = {
+                            page: request.query.page,
+                            count: request.query.count
+                        };
+                    }
```

Comment:
Can we have a default `config.paginate` in the case that client never specifically mention about pagination? This is related to my comment on the [models PR](https://github.com/screwdriver-cd/models/pull/619#discussion_r1540191979).